File: Segundo/2.Ejercicio.py
# -*- coding: utf-8 -*-
"""
@author: NOEMI
"""

def calculadora(o,a,b):
    suma = lambda x,y: x+y
    resta = lambda x,y: x-y
    multiplicacion = lambda x,y: x*y
    division = lambda x,y: x/y
    switcher = {
        "+" : suma(a,b),
        "-" : resta(a,b),
        "*" : multiplicacion(a,b),
        "/" : division(a,b), 
    }
    return switcher.get(o)
    

a = int(input("Primer Numero Entrada: "))
b = int(input("Segundo Numero Entrada: "))
print("Suma:",calculadora("+",a,b))
print("Resta:",calculadora("-",a,b))
print("Multiplicacion:",calculadora("*",a,b))
print("Division:",calculadora("/",a,b))

File: Segundo/2.Ejercicio.scala
object Primero {
   def main(args: Array[String]) {
        print("Primero numero: ")
        var a = scala.io.StdIn.readInt()
        print("Segundo numero: ")
        var b = scala.io.StdIn.readInt()
        println("Suma: " + calculadora("+",a,b));
        println("Resta: " + calculadora("-",a,b));
        println("Multiplicacion: " + calculadora("*",a,b));
        println("Division: " + calculadora("/",a,b));
        
   }
   def calculadora(x: String, y: Int, z: Int): Int = {
        x match { 
            case "+" => ((a: Int, b:Int) => {a+b})(y,z)
            case "-" => ((a: Int, b:Int) => {a-b})(y,z)
            case "*" => ((a: Int, b:Int) => {a*b})(y,z)
            case "/" => ((a: Int, b:Int) => {a/b})(y,z)
        }
    }
}